tests/platform_helper/test_command_environment.py

```
@@ -329,28 +327,94 @@ class TestGenerate:
+
         mocked_session = MagicMock()
         mock_get_aws_session.return_value = mocked_session
         fakefs.add_real_directory(
             BASE_DIR / "tests" / "platform_helper", read_only=False, target_path="copilot"
         )
+        fakefs.create_file(
+            PLATFORM_CONFIG_FILE,
+            contents=yaml.dump({"application": "my-app", "environments": environment_config}),
+        )
+        if is_terraform:
+            fakefs.create_dir("./terraform")
+
         result = CliRunner().invoke(generate, ["--name", "test"])
 
         actual = yaml.safe_load(Path("copilot/environments/test/manifest.yml").read_text())
         expected = yaml.safe_load(
             Path("copilot/fixtures/test_environment_manifest.yml").read_text()
         )
 
-        mock_get_vpc_id.assert_called_once_with(mocked_session, "test", None)
+        mock_get_vpc_id.assert_called_once_with(mocked_session, "test", expected_vpc)
         mock_get_subnet_ids.assert_called_once_with(mocked_session, "vpc-abc123")
         mock_get_cert_arn.assert_called_once_with(mocked_session, "test")
+
         assert actual == expected
         assert "File copilot/environments/test/manifest.yml created" in result.output
+        if is_terraform:
+            assert "File terraform/environments/test/main.tf created" in result.output
+        else:
+            assert "File terraform/environments/test/main.tf created" not in result.output
+
+    def test_fail_early_if_platform_config_invalid(self, fakefs):
+        from dbt_platform_helper.commands.environment import generate
+
+        fakefs.add_real_directory(
+            BASE_DIR / "tests" / "platform_helper", read_only=False, target_path="copilot"
+        )
+        fakefs.create_file(PLATFORM_CONFIG_FILE, contents=yaml.dump({}))
+
+        result = CliRunner().invoke(generate, ["--name", "test"])
+
+        assert result.exit_code != 0
+        assert "Missing key: 'application'" in result.output
```

Is this the only thing that's actually considered invalid?

Or are other invalinesses handled elsewhere?

---

It's the only thing we care about in this test, yes. I wanted to move the config validation earlier in the command so that the command fails early if the config is invalid. So a missing key achieves that. test_validation.py is where the comprehensive config validation takes place.